flake.nix

```diff
@@ -0,0 +1,118 @@
+{
+  description = "Hyperspace";
+
+  inputs.nixpkgs.url = "nixpkgs/nixpkgs-unstable";
+
+  outputs = { self, nixpkgs }:
+    let
+      supportedSystems =
+        [ "x86_64-linux" "x86_64-darwin" "i686-linux" "aarch64-linux" ];
+      forAllSystems = f:
+        nixpkgs.lib.genAttrs supportedSystems (system: f system);
+      packageJson = builtins.fromJSON (builtins.readFile ./package.json);
+
+    in {
+      overlay = final: prev: {
+        hyperspace = final.mkYarnPackage {
+          name = packageJson.name;
+          src = ./.;
+          packageJSON = ./package.json;
+          yarnLock = ./yarn2nix.lock;
+        };
+      };
+
+      defaultPackage = forAllSystems (system:
+        (import nixpkgs {
+          inherit system;
+          overlays = [ self.overlay ];
+        }).hyperspace);
+
+      apps = forAllSystems (system: {
+        refresh-dependencies = let pkgs = import nixpkgs { inherit system; };
+        in {
+          type = "app";
+          program = "${
+              pkgs.writeShellScriptBin "refresh-dependencies" ''
+                ${pkgs.nodejs}/bin/npm install --package-lock-only;
+                ${pkgs.nodePackages.yarn}/bin/yarn import;
+                rm package-lock.json;
+                mv yarn.lock yarn2nix.lock;''
+            }/bin/refresh-dependencies";
+        };
+      });
+
+      nixosModules.hyperspace = { config, pkgs, lib, ... }:
+        let cfg = config.services.hyperspace;
+
+        in with lib; {
+          options = {
+            services.hyperspace = {
+              enable = mkEnableOption "hyperspace";
+
+              storage = mkOption {
+                type = types.str;
+                default = "~/.hyperspace";
+                description = "Hyperspace storage directory";
+              };
+
+              host = mkOption {
+                type = types.str;
+                default = "hyperspace";
+                description = "Set unix socket name";
+              };
+
+              port = mkOption {
+                type = types.port;
+                default = 0;
+                description = "Set the port (will use TCP)";
```

Comment:
Maybe "If port set to 0 (default), uses TCP" to explain that subtle behavior more